implementation/src/main/java/io/smallrye/config/lifecycle/ConfigSourceInterceptorContext.java

```diff
@@ -0,0 +1,35 @@
+package io.smallrye.config.lifecycle;
+
+public interface ConfigSourceInterceptorContext {
+    /**
+     * This will provide the ConfigSourceInterceptor with the lookup value of the key on a ConfigSource.
+     *
+     * Should the ConfigSource be exposed?
+     *
+     * @return a ConfigValue
+     */
+    ConfigValue getConfigValue();
+
+    /**
+     * Lookup a key only into the current or previously evaluated ConfigSources
+     *
+     * @param key the key to lookup.
+     * @return the lookup key value.
+     */
+    String getValue(String key);
```

Comment:
This would be dropped.

Reply:
The idea with this was to plug it in the implementation to lookup additional keys into the previous iterated ConfigSources in the interceptor code. This will allow you to compose keys with the ConfigSource stack.